featuregates/client/pkg/featuregateclient/client.go

```diff
@@ -45,133 +46,217 @@ type Option func(*FeatureGateClient) *FeatureGateClient
 		return nil, fmt.Errorf("unable to create cluster client: %w", err)
 	}
 	return crClient, nil
 }
 
-// GetFeatureGate fetches the FeatureGate
-func (f *FeatureGateClient) GetFeatureGate(ctx context.Context, featureGateName string) (*configv1alpha1.FeatureGate, error) {
-	gate := &configv1alpha1.FeatureGate{}
-
-	if err := f.c.Get(ctx, client.ObjectKey{
-		Name: featureGateName,
-	}, gate); err != nil {
+// GetFeatureGate fetches the specified FeatureGate resource.
+func (f *FeatureGateClient) GetFeatureGate(ctx context.Context, featureGateName string) (*core.FeatureGate, error) {
+	gate := &core.FeatureGate{}
+	err := f.crClient.Get(ctx, client.ObjectKey{Name: featureGateName}, gate)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, ErrTypeNotFound
+		}
 		return nil, fmt.Errorf("couldn't get featuregate %s: %w", featureGateName, err)
 	}
 	return gate, nil
 }
 
-// GetFeature fetches the Feature
-func (f *FeatureGateClient) GetFeature(ctx context.Context, featureName string) (*configv1alpha1.Feature, error) {
-	feature := &configv1alpha1.Feature{}
+// GetFeatureGateList fetches all featuregates on the cluster.
+func (f *FeatureGateClient) GetFeatureGateList(ctx context.Context) (*core.FeatureGateList, error) {
+	gates := &core.FeatureGateList{}
+	err := f.crClient.List(ctx, gates)
+	if err != nil {
+		return nil, fmt.Errorf("get featuregates on cluster: %w", err)
+	}
+	return gates, nil
+}
 
-	if err := f.c.Get(ctx, client.ObjectKey{Name: featureName}, feature); err != nil {
+// GetFeature fetches the specified Feature resource.
+func (f *FeatureGateClient) GetFeature(ctx context.Context, featureName string) (*core.Feature, error) {
+	feature := &core.Feature{}
+	err := f.crClient.Get(ctx, client.ObjectKey{Name: featureName}, feature)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, ErrTypeNotFound
+		}
 		return nil, err
 	}
 	return feature, nil
 }
 
-// ActivateFeature activates a Feature
-func (f *FeatureGateClient) ActivateFeature(ctx context.Context, featureName, featureGateName string) error {
+// ActivateFeature activates a Feature if it passes validation and warranty checks.
+// Warning: Before sending `true` via the warrantyVoidAllowed function argument, ensure
+// explicit user awareness and approval if activating a Feature will cause the support
+// warranty to be void. Once warranty is void, it is is permanent for the environment.
+func (f *FeatureGateClient) ActivateFeature(ctx context.Context, featureName string, warrantyVoidAllowed bool) error {
+	// A Feature must exist in the cluster to be activated.
 	feature, err := f.GetFeature(ctx, featureName)
 	if err != nil {
-		return fmt.Errorf("couldn't get feature %s: %w", featureName, err)
+		return fmt.Errorf("get Feature %s: %w", featureName, err)
+	}
+
+	gates, err := f.GetFeatureGateList(ctx)
+	if err != nil {
+		return fmt.Errorf("get FeatureGateList: %w", err)
+	}
+
+	gateName, featRef := FeatureRefFromGateList(gates, featureName)
```

Comment:
same comment as above you can use the existing functions `GetFeatureGateForFeature` and `GetFeatureReferenceFromFeatureGate` from util package

Reply:
And again. I don't want to get FeatureGate using `GetFeatureGateForFeature`. I want to find the FeatureRef from a list of Feature Gates. It's backwards.

Reply:
Refer to previous mention of this comment for decision and resolution.